v2ray, qv2ray, v2raya: Update

r-ryantm opened the pull request:
Automatic update generated by [nixpkgs-update](https://github.com/ryantm/nixpkgs-update) tools. This update was made based on information from passthru.updateScript.


meta.description for v2ray is: A platform for building proxies to bypass network restrictions

meta.homepage for v2ray is: https://www.v2fly.org/en_US/




###### Updates performed
- Ran passthru.UpdateScript

###### To inspect upstream changes

- [Release on GitHub](https://github.com/v2fly/v2ray-core/releases/tag/v5.8.0)

- [Compare changes on GitHub](https://github.com/v2fly/v2ray-core/compare/v5.7.0...v5.8.0)

###### Impact

<details>
<summary>
<b>Checks done</b> (click to expand)
</summary>

---

- built on NixOS
- The tests defined in `passthru.tests`, if any, passed
- found 5.8.0 with grep in /nix/store/82a1fch8kpn58s88c377vhg3n9f2hz8m-v2ray-core-5.8.0
- found 5.8.0 in filename of file in /nix/store/82a1fch8kpn58s88c377vhg3n9f2hz8m-v2ray-core-5.8.0

---

</details>
<details>
<summary>
<b>Rebuild report</b> (if merged into master) (click to expand)
</summary>

```
3 total rebuild path(s)

3 package rebuild(s)

First fifty rebuilds by attrpath
qv2ray
v2ray
v2raya
```

</details>

<details>
<summary>
<b>Instructions to test this update</b> (click to expand)
</summary>

---

*(Removed as this is out-of-date)*

</details>

### Pre-merge build results

*(Removed as this is out-of-date)*

---

###### Maintainer pings

cc @servalcatty for [testing](https://github.com/ryantm/nixpkgs-update/blob/master/doc/nixpkgs-maintainer-faq.md#r-ryantm-opened-a-pr-for-my-package-what-do-i-do).

Also ping qv2ray maintainers @poscat0x04 @wineee , v2raya maintainers @Eliot00

Eliot00 committed:
v2raya: 2.0.5 -> 2.2.4

https://github.com/v2rayA/v2rayA/compare/v2.0.5...v2.2.4

This fixes startup issue with go 1.21, see upstream issue 1017.

Co-authored-by: Bobby Rong <rjl931189261@126.com>

r-ryantm committed:
v2ray: 5.7.0 -> 5.8.0

https://github.com/v2fly/v2ray-core/compare/v5.7.0...v5.8.0

Fixes https://hydra.nixos.org/build/239295592/nixlog/1

1sixth reviewed (approved):
`v2ray` fails to build on master right now. This PR fixes it.

bobby285271 commented:
@ofborg build v2ray v2ray.passthru.tests

(v2ray-geoip was fixed in 695e41565eb458417809f64e7d4008168e5df4d3)

bobby285271 commented:
@ofborg build qv2ray

bobby285271 committed:
qv2ray: unstable-2023-06-09 -> unstable-2023-07-11

https://github.com/Qv2ray/Qv2ray/commit/b3080564809dd8aef864a54ca1b79f0984fe986b

Fixes:

/build/source/src/base/Qv2rayLog.hpp:17:29: error: expected primary-expression before ',' token
   17 | #define ___LOG_EXPAND(___x) , QPair<std::string, decltype(___x)>(std::string(#___x), [&] { return ___x; }())
      |                             ^
/build/source/src/base/Qv2rayLog.hpp:17:63: error: expected primary-expression before ')' token
   17 | #define ___LOG_EXPAND(___x) , QPair<std::string, decltype(___x)>(std::string(#___x), [&] { return ___x; }())
      |                                                               ^
/build/source/src/base/Qv2rayLog.hpp:17:64: error: template argument 2 is invalid
   17 | #define ___LOG_EXPAND(___x) , QPair<std::string, decltype(___x)>(std::string(#___x), [&] { return ___x; }())
      |                                                                ^

bobby285271 commented:
Sounds like the go bump not only causes build issues, but also startup issue for v2raya

bobby285271 commented:
Merging early since all three are already failing on Hydra (due to v2ray).
